app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt

```diff
@@ -287,7 +278,33 @@ class TabPreferenceActivity : BaseActivity(), Injectable, ItemInteractionListene
                 updateAvailableTabs()
                 saveTabs()
             }
-            .show()
+
+        if (adapter.isEmpty) {
+            dialogBuilder.setTitle(R.string.select_list_title_empty)
+            dialogBuilder.setNeutralButton(R.string.select_list_create) { _, _ ->
+                val listIntent = Intent(applicationContext, ListsActivity::class.java)
+                startActivity(listIntent)
+            }
+        }
+
+        val dialog = dialogBuilder.show()
+
+        lifecycleScope.launch {
+            mastodonApi.getLists().fold(
+                { lists ->
+                    adapter.addAll(lists)
+                    if (lists.isNotEmpty()) {
+                        dialog.getButton(DialogInterface.BUTTON_NEUTRAL).hide()
```

Comment:
Not really sure this is a good workflow: For slow loads the gui will change noticably.
For very slow loads it will be wrong.

However also a "still loading information" progress animation is missing here.

Reply:
Why not put *all* the dialog code into a function that gets called from the success block of `fold`?

Reply:
What would happen inbetween? (just to clarify) So you click "add", nothing happens (loading api items in the background) and then the dialog appears?

Reply:
(added now a loading progress indicator)